docs/content/tutorial_code/docker.md

```diff
@@ -0,0 +1,146 @@
+---
+title: "Envoy WASM in docker"
+weight: 2
+description: Run/Test your WebAssembly filter locally with Envoy in docker
+---
+
+`wasme` makes creating Envoy filters easier than ever, now the question becomes how do we run and test them.
+This tutorial will explain in detail how to run and test Envoy wasm filters locally, before deploying them.
+
+For the purposes of this example we will be using a simple stats filter `webassemblyhub.io/yuval-k/metrics:v1`. However, 
+the techniques used here can be applied to any and all wasm filters built this way.
+
+#### Prerequisites
+* [docker](https://www.docker.com/)
+* [wasme](https://github.com/solo-io/wasme)
+
+## Configuration
+
+The first step in testing the filter locally is creating the static E config which will serve out new filter.
+Documentation on E's API can be found [here](https://www.envoyproxy.io/docs/envoy/v1.12.0/api/api). 
+
+Below is the full yaml necessary to test. We will unpack it step by step after.
+```yaml
+admin:
+  access_log_path: /dev/null
+  address:
+    socket_address:
+      address: 0.0.0.0
+      port_value: 19001
+static_resources:
+  listeners:
+  - name: listener_0
+    address:
+      socket_address: { address: 0.0.0.0, port_value: 8080 }
+    filter_chains:
+    - filters:
+      - name: envoy.http_connection_manager
+        config:
+          codec_type: auto
+          stat_prefix: http
+          route_config:
+            name: test
+            virtual_hosts:
+            - name: test
+              domains: ["*"]
+              routes:
+              - match: { prefix: "/" }
+                route: { cluster: static-cluster }
+          http_filters:
+          - name: envoy.filters.http.wasm
+            config: 
+              config: 
+                name: "test"
+                root_id: "stats_root_id"
+                vm_config:
+                  vm_id: test
+                  runtime: envoy.wasm.runtime.v8
+                  allow_precompiled: false
+                  code:
+                    local:
+                      filename: /etc/filter.wasm
+                configuration: |
+                  {}
+          - name: envoy.router
+  clusters:
+  - name: static-cluster
+    connect_timeout: 0.25s
+    type: logical_dns
+    lb_policy: round_robin
+    hosts: [{ socket_address: { address: host.docker.internal, port_value: 10101 } }]
+```
+
+The first part of the config is not particularly interesting. The admin section simply tells Envoy which port to use for 
+the admin interface.
```

Comment:
weird line break

Reply:
Breaks like that don't actually show up in the rendered docs. I just did it because I don't like having really long lines